independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CurrentManagedContext.java

```diff
@@ -271,6 +290,14 @@ public boolean isValid() {
             return isValid == VALID;
         }
 
+        boolean shouldFireInitializedEvent() {
+            return INITIALIZED_FIRED.compareAndSet(this, FALSE, TRUE);
+        }
+
+        public boolean shouldFireBeforeDestroyedEvent() {
```

Comment:
Why `public`? Or why is it different compared to `shouldFireInitializedEvent()`?

Reply:
They used to be both `public` but I forgot to remove it from this one - will do.